### PR TITLE
MP serverlist GUI now fully functional

### LINKED ISSUE #1193: In-game serverlist not loading
The in-game server list isn't woking yet, I would like to see it finished before releasing 0.4.7.
The server list is already able to return the list in JSON: http://multiplayer.rigsofrods.org/server-list?json=true

### PR DESCRIPTION
**Pull request by only-a-ptr**
ror-serverlist-yay

It all works!

Fixes #1193

### CODE REVIEW

**Comment by ulteq** [comment]
Do we really need to add almost 10.000 lines for this?

> Only manual refreshing works.

I think that's just fine. I don't see the purpose of auto refreshing.

**Comment by only-a-ptr** [comment]
> Do we really need to add almost 10.000 lines for this?

Most of that is JsonCpp library (to be moved out to deps) and the cloned/patched MyGUI::MultiListBox

Initially I thought I'll handle HTTP with rorserver code+SocketW, so I incorporated JsonCpp for simplicity. But now that I had to use C++Requests library externally, update of our deps is needed anyway, so we can move JsonCpp out right away.

> I think that's just fine. I don't see the purpose of auto refreshing

We don't need periodic auto-refresh, but IMO it should update once when opened.

**Comment by ulteq** [comment]
> I meant it doesn't auto-update when opened.

Ok, that should be changed.

**Comment by tritonas00** [comment]
Great, but this PR is extremely linux unfriendly to compile

https://paste.ee/p/yEQ7s

**Comment by only-a-ptr** [comment]
> Great, but this PR is extremely linux unfriendly to compile

It appears to be some hassle between Linux and libcurl. They somehow fight about OS headers.

File: source/main/utils/Utils.cpp [hunk 1]
```diff
@@ -28,7 +28,9 @@
 #include <Ogre.h>
 
 #ifndef _WIN32
-#include <iconv.h>
+#   include <iconv.h>
+#else
+#   include <windows.h> // Sleep()
```
Comment (ulteq) [inline source/main/utils/Utils.cpp]: Typo?

File: source/main/gui/panels/GUI_MultiplayerSelector.cpp [hunk 2]
```diff
@@ -35,94 +35,246 @@
 #include "GUIManager.h"
 #include "Application.h"
 #include "MainThread.h"
+#include "Http.h"
```
Comment (ulteq) [inline source/main/gui/panels/GUI_MultiplayerSelector.cpp]: Do we still need this?

File: source/main/main.cpp [hunk 3]
```diff
@@ -45,6 +45,10 @@ using namespace Ogre;
 #include "ShellAPI.h"
 #endif //OGRE_PLATFORM_WIN32
 
+#ifdef USE_CURL
+#   include <curl/curl.h>
+#endif //USE_CURL
```
Comment (ulteq) [inline source/main/main.cpp]: Do we need `include <stdio.h>` here?

File: source/main/CMakeLists.txt [hunk 4]
```diff
@@ -410,6 +425,7 @@ target_include_directories( ${BINNAME} PRIVATE
  threadpool
  utils
  utils/profiler
+ utils/json
```
Comment (ulteq) [inline source/main/CMakeLists.txt]: This line seems to break the build process on linux.

File: source/main/gui/panels/GUI_MultiplayerSelector.cpp [hunk 5]
```diff
@@ -35,94 +35,246 @@
...
 
+#include "json.h"
```
Comment (ulteq) [inline source/main/gui/panels/GUI_MultiplayerSelector.cpp]: Only `#include "json/json.h"` seems to work on linux.

File: source/main/main.cpp (hunk 3 above)
Comment (only-a-ptr) [inline source/main/main.cpp]: I don't know. But older includes of curl.h were always coupled with stdio.h

**Comment by only-a-ptr** [comment]
@ulteq I squashed your build fix and CMake alphabetic cleanup

**Comment by tritonas00** [comment]
Now compiles and works fine  :)

**Pull request merged**